py/plugins/firebase/src/genkit/plugins/firebase/constant.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from enum import StrEnum
+from typing import Any, Callable, Dict, Optional
+
+from google.cloud import firestore
+from google.cloud.firestore_v1.base_vector_query import DistanceMeasure
+from pydantic import BaseModel
+
+
+class FirestoreRetrieverConfig(BaseModel):
+    """The name of the retriever."""
+
+    name: str
+    """The name of the Firestore collection to query."""
+    collection: str
+    """The name of the field containing the vector embeddings."""
+    vector_field: str
+    """The name of the field containing the document content, you wish to return."""
+    content_field: str
+    """The embedder to use with this retriever."""
+    embedder: str
+    """The distance measure to use when comparing vectors. Defaults to 'COSINE'."""
+    distance_measure: DistanceMeasure = DistanceMeasure.COSINE
+    """The Firestore database instance from which to query."""
+    firestore_client: Any
+    """Optional list of metadata fields to include."""
+    metadata_fields: Optional[list[str]] = None
```

Comment:
curious why no support for the transform function?

https://github.com/firebase/genkit/blob/ede71f43f0665369aff2845ee06bc64a463840ef/js/plugins/firebase/src/firestoreRetriever.ts#L115